src/components/workspace-utils.js

```diff
@@ -38,6 +38,25 @@ export const useWorkspaces = () => {
   return { workspaces, refresh, loading }
 }
 
+export const useWorkspaceDetails = ({ namespace, name }, fields) => {
```

Comment:
It looks like we have a variety of components that are calling `await Ajax(signal).Workspaces.workspace(namespace, name).details`. Could this new code be used in those places as well to help clean things up? 

FYI @dmohs -- this PR is a follow-up from one you reviewed, so it might make sense for you to review it. Also, you might have a perspective on how this should be factored.

Reply:
I think so. On the other hand, using `useThing` hooks for API requests might be undesirable for components that make multiple API requests if each hook has its own cancellation signal and busy state vs sharing one for all API requests.

Reply:
I looked through the other calls to `details` and only saw one that was also making the call `onMount`. I went ahead and replaced it-- let me know what you think.